src/containers/add-resources/AddResources.tsx

```diff
@@ -0,0 +1,136 @@
+import { useCallback, useState } from 'react'
+
+import { useSnackBarContext } from '~/context/snackbar-context'
+import { useModalContext } from '~/context/modal-context'
+import useSelect from '~/hooks/table/use-select'
+import useSort from '~/hooks/table/use-sort'
+import useAxios from '~/hooks/use-axios'
+import useBreakpoints from '~/hooks/use-breakpoints'
+import { AxiosResponse } from 'axios'
+
+import AddResourceModal from '~/containers/my-resources/add-resource-modal/AddResourceModal'
+
+import {
+  columns,
+  initialSort,
+  removeColumnRules
+} from '~/containers/add-resources/AddResources.constants'
+import { defaultResponses, snackbarVariants } from '~/constants'
+
+import { ajustColumns } from '~/utils/helper-functions'
+import {
+  ErrorResponse,
+  GetResourcesParams,
+  ItemsWithCount,
+  CourseResources
+} from '~/types'
+
+type GetServiceFunction<GetResourcesParams, T> = (
+  params?: GetResourcesParams
+) => Promise<AxiosResponse<ItemsWithCount<T>>>
+
+interface AddResourcesProps<T> {
+  resources: T[]
+  onAddResources: (resource: T[]) => void
+  resourceType: string
+  requestService: GetServiceFunction<GetResourcesParams, T>
+}
+
+const AddResources = <T extends CourseResources>({
```

Comment:
It is great that you want to reuse this component for adding all resources, so, I guess you need to remove AddAttachment and AddQuestions components and replace them with this component